src/derivative_operators/derivative_operator_functions.jl

```diff
@@ -0,0 +1,40 @@
+for MT in [AbstractArray, AbstractMatrix]
```

Comment:
Why have a separate one for AbstractMatrix? AbstractMatrix == AbstractArray{T,2}

Reply:
I was getting ambiguity issues with 
`mul!(x_temp::AbstractArray{T,2}, A::AbstractDerivativeOperator{T}, M::AbstractMatrix{T})`

Reply:
Since `AbstractDerivativeOperator{T}` doesn't specify the axis of differentiation I had to specify `A` as being a `DerivativeOperator{T,N}`

Reply:
I could use something like `typeof(A).parameters[2]` for `mul!(x_temp::AbstractArray{T}, A::AbstractDerivativeOperator{T}, M::AsbtractArray{T}) where T`, but I don't know if this is okay with the generality of `AbstractDerivativeOperator` that we might want to maintain.